### PR TITLE
Bot using Sharepoint List : csharp

### PR DESCRIPTION
**Pull request by Trinetra-MSFT**
## Description 

This app demonstrates how a bot communicates in teams and stores the conversation data in SharePoint List.
Bot uses the a SharePoint App to perform the `CRUD `operations in the SharePoint List

**language used:**  C#

## Testing
**Upload the custom app in Teams**
image

**Interaction with the Bot**
image

**Ping the Bot**
image

### CODE REVIEW

File: samples/bot-sharepoint-list/csharp/SPFx_Bot/teamsAppManifest/manifest.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        "identity",
+        "messageTeamMembers"
+    ],
+    "validDomains": [
+        "xxxxx.sharepoint.com",
+        "*.ngrok.io",
+        "token.botframework.com"
+    ],
+    "webApplicationInfo": {
+        "id": "00000003-0000-0ff1-ce00-000000000000",
```
Comment (Wajeed-msft) [inline samples/bot-sharepoint-list/csharp/SPFx_Bot/teamsAppManifest/manifest.json:62]: webApplicationInfo is not needed. Please remove.

File: samples/bot-sharepoint-list/csharp/SPFx_Bot/teamsAppManifest/manifest.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    "description": {
+        "short": "SPFx app for Microsoft Teams",
+        "full": "This app will save conversation data to sharepoint list"
+    },
+    "accentColor": "#FFFFFF",
+    "staticTabs": [
+        {
+            "entityId": "conversations",
+            "scopes": [
```
Comment (Wajeed-msft) [inline samples/bot-sharepoint-list/csharp/SPFx_Bot/teamsAppManifest/manifest.json:29]: staticTabs are not required. Please delete.

File: samples/bot-sharepoint-list/csharp/SPFx_Bot/Repositories/SharepointRepository.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            HttpResponseMessage response = await httpclient.SendAsync(request);
+            string responseBody = await response.Content.ReadAsStringAsync();
+
+            if (!response.IsSuccessStatusCode)
+                throw new Exception(responseBody);
+
+            string accessToken = null;
+            accessToken = JsonConvert.DeserializeObject<SharepointRepository.TokenResponse>(responseBody).access_token;
+            return accessToken;
+        }
```
Comment (Wajeed-msft) [inline samples/bot-sharepoint-list/csharp/SPFx_Bot/Repositories/SharepointRepository.cs:42]: Please add new line after function call. Check other classes and make same changes.

File: samples/bot-sharepoint-list/csharp/SPFx_Bot/Bots/SpfxBot.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+using Microsoft.BotBuilderSamples.SpfxBot.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.BotBuilderSamples.SpfxBot.Bots
+{
+    public class SpfxBot : ActivityHandler
```
Comment (Wajeed-msft) [inline samples/bot-sharepoint-list/csharp/SPFx_Bot/Bots/SpfxBot.cs:17]: Please rename this to SharePointListBot. 
SPFx is a page and web part model that provides full support for client-side SharePoint development We are just using SharePoint list.

File: README.md [hunk 5]
```diff
@@ -37,6 +37,7 @@
...
 |7| Task Module | Demonstrating how to retrieve a Task Module and values from cards in it, for a Messaging Extension. | [View](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/54.teams-task-module) | [View](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/javascript_nodejs/54.teams-task-module) | [View](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/54.teams-task-module) |
 |8| Start new thread in a channel | Demonstrating how to create a new thread in a channel. | [View](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/58.teams-start-new-thread-in-channel) | [View](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/javascript_nodejs/58.teams-start-new-thread-in-channel) | [View](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/58.teams-start-thread-in-channel) |
+|9| Bot using Sharepoint List | This app demonstrate how a Bot saves data in sharepoint list | [View][bot-sharepoint-list#cs] | [View]() | [View]() |
```
Comment (Wajeed-msft) [inline README.md:40]: Could you please move this under https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/v-trkum/SPFxbot#additional-samples?

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

**Review by Wajeed-msft — APPROVED** [review]

**Review by asnow003 — APPROVED** [review]

**Comment by Trinetra-MSFT** [comment]
Merging this Pull Request as changes are approved

**Pull request merged**